globus_compute/gem_globus_script.py

```diff
@@ -35,21 +49,51 @@ def run_gem_reconnection():
     with open(error_file, "r") as err_file:
         job_error = err_file.read()
 
-    return job_output, job_error
+    if "Main loop completed" in job_output:
+        return job_output, job_error, 0
+    else:
+        return job_output, job_error, 1
+
+# Reset output files
+if os.path.isfile("gem_job.out"):
+        os.remove("gem_job.out")
+if os.path.isfile("gem_job.err"):
+        os.remove("gem_job.err")
 
 # Submit to Globus
 print("Starting script")
-client = Client()
-anvil_endpoint_id = os.environ.get('ANVIL_ENDPOINT_ID')
-with Executor(endpoint_id=anvil_endpoint_id) as ex:
-    print("Submitting function")
-    future = ex.submit(run_gem_reconnection)
+try:
+    client = Client()
+    
+    anvil_endpoint_id = os.environ.get('ANVIL_ENDPOINT_ID')
+    if not anvil_endpoint_id:
+        raise ValueError("Environment variable 'ANVIL_ENDPOINT_ID' is not set.")
+    
+    with Executor(endpoint_id=anvil_endpoint_id) as ex:
+        print("Submitting function")
+        future = ex.submit(run_gem_reconnection)
 
-    out, err = future.result()
-    with open("gem_job.out", "w") as out_file:
-        out_file.write(out)
+        out, err, status = future.result()
+        with open("gem_job.out", "w") as out_file:
+            out_file.write(out)
 
+        with open("gem_job.err", "w") as err_file:
+            err_file.write(err)
+except Exception as e:
+    print("Error occured while submitting to Globus")
     with open("gem_job.err", "w") as err_file:
-        err_file.write(err)
+        err_file.write(f"Error: {e}")
+    status = 4
 
-print("Output and error files have been saved")
+# Print results
+print(f"Exit code {status}: ", end="")
+if status == 0:
+    print("Job completed successfully.")
+elif status == 1:
+    print("Job was submitted but did not complete successfully.")
+elif status == 2:
+    print("Slurm error encountered while submitting the job.")
+elif status == 3:
+    print("Job timed out during execution.")
+elif status == 4:
```

Comment:
Would you please use a python enum (or similar) instead of 'hard coded' integers for the status flags to improve readability?

Reply:
I added a commit with that change.